docs/tutorial/go.rst

```diff
@@ -0,0 +1,361 @@
+Rockcraft will compile the Go application. Run the Go application using
+``./go-hello-world`` to verify that it works.
+
+The code in ``main.go`` starts up an HTTP server listening on port 8000
+that we can test by using ``curl`` to send a request to the root
+endpoint. We may need a new terminal for this, if we are using Multipass use
+``multipass shell rock-dev`` to get another terminal:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:curl-go]
+    :end-before: [docs:curl-go-end]
+    :dedent: 2
+
+The Go application should respond with ``Hello, world!``.
+
+The Go application looks good, so stop it for now using :kbd:`Ctrl` + :kbd:`C`.
+
+Pack the Go application into a rock
+===================================
+
+
+First, we'll need a ``rockcraft.yaml`` file. Rockcraft will automate its
+creation and tailor it for a Go application by using the
+``go-framework`` profile:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+The ``rockcraft.yaml`` file will automatically be created in the project working
+directory. Open it in a text editor and check that the ``name`` is
+``go-hello-world``. Ensure that ``platforms`` includes the architecture of
+the host. For example, if the host uses the ARM
+architecture, include ``arm64`` in ``platforms``.
+
+.. note::
+    For this tutorial, we'll use the ``name`` ``go-hello-world`` and assume
+    we are running on an ``amd64`` platform. Check the architecture of the
+    system using ``dpkg --print-architecture``.
+
+    The ``name``, ``version`` and ``platform`` all influence the name of the
+    generated ``.rock`` file.
+
+Pack the rock:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:pack]
+    :end-before: [docs:pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on the network, this step can take a couple of minutes to finish.
+
+    ``ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`` is required whilst the ``go-framework``
+    extension is experimental.
+
+Once Rockcraft has finished packing the Go rock, we'll find a new file in
+the working directory (an `OCI <OCI_image_spec_>`_ archive) with the ``.rock``
+extension:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:ls-rock]
+    :end-before: [docs:ls-rock-end]
+    :dedent: 2
+
+.. note::
+    If we changed the ``name`` or ``version`` in ``rockcraft.yaml`` or are not
+    on an ``amd64`` platform, the name of the ``.rock`` file will be different.
+
+    By default, the ``go-framework`` will use the ``bare`` base.
+    The ``bare`` base is recommended for production applications because
+    the image size will be smaller, and the risk of security issues lower.
+    However, we may want to change the base to ``ubuntu@24.04`` during the
+    development process to benefit from the extra utilities provided.
```

Comment:
This is all supplemental information more suited toward a how-to or reference, especially when we consider that later we specifically mention the resulting rock name as `go-hello-world_0.1_amd64.rock`.

If you're amenable, I'd like to remove it.

Reply:
sure, removed the full note.